### PR TITLE
added json support for insert_document_from_file

### PR DESCRIPTION
**Pull request by edgargarciamacrometa**
## Description

Added json support for insert_document_from_file, also added files to test in the files directory.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Reviews

Please identify two developers to review this change

- [x] @koshy-thomas 
- [ ] @dlozina-macrometa 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added new functional tests that prove my fix is effective or that my feature works
- [x] Existing and new functional tests pass with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: c8/collection.py [hunk 1]
```diff
@@ -988,21 +989,35 @@ def insert_from_file(self, csv_filepath, return_new=False, sync=None, silent=Fal
...
+                        data_dict.setdefault(column, {index: value})
+                        temp_dict = data_dict.get(column)
+                        temp_dict.update({index: value})
+                    index += 1
+
+                documents, index = self.get_documents_from_file(data_dict, 0)
+                resp = self.insert_many(documents, return_new, sync, silent)
+                result.append(resp)
+            except StopIteration:
+                raise Exception("Invalid CSV file")
```
Comment (dlozina-macrometa) [inline c8/collection.py:1010]: todo

File: tests/test_collection.py [hunk 2]
```diff
@@ -132,3 +135,42 @@ def test_collection_management(sys_fabric, client, bad_fabric):
...
+    entries = ("_id", "_key", "_rev")
+    for doc in data:
+        for key in entries:
+            if key in doc:
+                del doc[key]
+
+    assert documents == data
+    col.truncate()
+
+    client.collection(col.name)
```
Comment (koshy-thomas) [inline tests/test_collection.py:159]: This line is not needed it seems?

**Comment by koshy-thomas** [comment]
Great Work! @edgargarciamacrometa just a very very minor comment

File: tests/test_collection.py (hunk 2 above)
Comment (dlozina-macrometa) [reply tests/test_collection.py:159]: @edgargarciamacrometa could you please check this out so we can merge if it is oke? cc @koshy-thomas

Comment (koshy-thomas) [reply tests/test_collection.py]: @dlozina-macrometa I have removed this line, resolved merge conflict and also fixed a test line which was left commented.. You can proceed with the merge

**Review by koshy-thomas — APPROVED** [review]

File: tests/test_collection.py (hunk 2 above)
Comment (dlozina-macrometa) [reply tests/test_collection.py]: @koshy-thomas that is great! Thanks

**Pull request merged**